libkbfs/bserver_remote.go

```diff
@@ -264,13 +264,16 @@ func (b *BlockServerRemote) Put(ctx context.Context, id BlockID, tlfID TlfID,
 		// BlockKey is misnamed -- it contains just the server
 		// half.
 		BlockKey: serverHalf.String(),
-		Folder:   tlfID.String(),
+		Folder:   md.ID.String(),
 		Buf:      buf,
 	}
 
 	err = b.client.PutBlock(ctx, arg)
 	if err != nil {
 		if qe, ok := err.(BServerErrorOverQuota); ok && !qe.Throttled {
+			name := md.GetTlfHandle().GetCanonicalName()
```

Comment:
I struggled with this. My issue with passing the name in that it's an extra step for `BlockOps` to take, that's completely useless 99.99% of the time.  A `TlfHandle` has the same issue, plus it's also slightly burdensome to construct during a test.  I wanted production code to only look up the name when it was necessary to do so -- putting it in the Put interface method seemed like it would be more confusing to someone trying to learn those interfaces.  Using `RootMetadata` kept the `BlockOps` code simple, and kept the number of parameters to one, and I thought that was more important than the test code.